tuning/autotune.py

```diff
@@ -50,12 +51,36 @@ class CandidateTracker:
     unet_vmfb_hash: Optional[str] = None
     unet_benchmark_time: Optional[float] = None
 
+def fetch_available_devices() -> list[any]:
+    drivers = ireert.HalDriver.query()
+    all_device_ids = []
 
-def parse_devices(devices_str: str) -> list[int]:
-    """Parse a comma-separated list of device IDs (e.g., "1,3,5" -> [1, 3, 5])."""
-    devices = []
+    for driver_name in drivers:
+        try:
+            driver = ireert.get_driver(driver_name)
+            devices = driver.query_available_devices()
+            all_device_ids.extend(f"{driver_name}://{device['path']}" for device in devices)
```

Comment:
We can filter the drivers based on the device prefix, e.g., `--devices=hip://0` would only look for hip devices and not print warnings about missing cuda drivers:
```
Could not initialize driver cuda: Error creating driver: iree/runtime/src/iree/hal/drivers/cuda/cuda_dynamic_symbols.c:66: UNAVAILABLE; CUDA driver library 'libcuda.so'/'nvcuda.dll' not available; please ensure installed and in dynamic library search path
```